src/pages/sale-zones/forms/SaleZoneForm.js

```diff
@@ -0,0 +1,169 @@
+// ** MUI Imports
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { useForm } from 'src/hooks/form'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { DataSets } from 'src/resources/DataSets'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+
+export default function SaleZoneForm({ labels, maxAccess, recordId }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.SalesZone.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      name: '',
+      szRef: '',
+      countryId: '',
+      parentId: '',
+      parentName: ''
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      name: yup.string().required(' '),
+      countryId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      try {
+        const recordId = obj.recordId
+
+        const response = await postRequest({
+          extension: SaleRepository.SalesZone.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!recordId) {
+          toast.success(platformLabels.Added)
+          formik.setValues({
+            ...obj,
```

Comment:
this can be replaced by
```
!recordId ? toast.success(platformLabels.Added) : toast.success(platformLabels.Edited)
formik.setFieldValue('recordId',response.recordId)
```